Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js

```diff
@@ -142,6 +165,87 @@ WI.FontVariationDetailsSectionRow = class FontVariationDetailsSectionRow extends
 
         return 1;
     }
+
+    _showValidity()
+    {
+        if (!this._hasValidValue)
+            this.warningMessage = WI.UIString("Axis value outside of supported range: %s – %s", "A warning that is shown in the Font Details Sidebar when the value for a variation axis is outside of the supported range of values").format(this._formatAxisValueAsString(this._minimumValue), this._formatAxisValueAsString(this._maximumValue));
+        else
+            this.warningMessage = null;
+    }
+
+    _handleValueBlurEvent(event)
+    {
+        this._skipUpdatingInputValue = false;
+        this.value = this._value;
+    }
+
+    _handleValueInputEvent(event)
+    {
+        if (this._showValidityTimeout)
+            clearTimeout(this._showValidityTimeout);
+
+        // Allow the input to temporarily hold an invalid value without immediately warning the user.
+        this._showValidityTimeout = setTimeout(this._showValidity.bind(this), 750);
+
+        let valueAsString = event.target.value;
+        let valueAsNumber = event.target.valueAsNumber;
+
+        this._hasValidValue = false;
+        const validValueFormat = /\-?\d+(.\d+)?/g;
+
+        if (!validValueFormat.test(valueAsString)) {
+            this._skipUpdatingInputValue = true;
+
+            this.dispatchEventToListeners(WI.FontVariationDetailsSectionRow.Event.VariationValueChanged, {tag: this._tag, value: this._defaultValue});
+            return;
+        }
+
+        if (!(this._minimumValue <= valueAsNumber && valueAsNumber <= this._maximumValue)) {
```

Comment:
we dont allow for `valueAsNumber === this._minimumValue` or `valueAsNumber === this._maximumValue`?

NIT: I'd avoid the `(!(` if possible as it's not as easy to read
```
if (valueAsNumber < this._minimumValue || valueAsNumber > this._maximumValue) {
```

Reply:
>we dont allow for valueAsNumber === this._minimumValue or valueAsNumber === this._maximumValue?

We do. The checks are "lower than or equal".

> NIT: I'd avoid the (!( if possible as it's not as easy to read
`if (valueAsNumber < this._minimumValue || valueAsNumber > this._maximumValue) {`

Agreed, this reads better. Thanks!